src/shogun/base/library.cpp

```diff
@@ -0,0 +1,99 @@
+#include <shogun/base/library.h>
+#include <shogun/lib/common.h>
+
+namespace shogun
+{
+    class LibraryHandle : public CSGObject
+    {
+    
+    public:
+        LibraryHandle(const std::string& filename)
+        {
+            dlerror();
+            handle = dlopen(filename.c_str(), RTLD_NOW | RTLD_LOCAL);
+            if (handle) 
+            {
+                SG_INFO("Loaded.\n");
+            }
+            else
+            {
+                SG_ERROR("Failed: %s.\n", dlerror());
+            }
+        }
+
+        ~LibraryHandle()
+        {
+            if (handle)
+            {
+                SG_INFO("Closing library.\n");
+                dlclose(handle);
+            }
+        }
+
+        template <typename T>
+        T call(const std::string& name)
+        {
+            dlerror();
+            T (*fm)();
+            *(void**)(&fm) = dlsym(handle, name.c_str());
+            char* potential_error = dlerror();
+            if (potential_error) 
```

Comment:
@lisitsyn I have changed a bit of error handling here based on http://linux.die.net/man/3/dlerror.